File: Leetcode/PY/Leetcode1185-一周中的第几天.py
class Solution:
    def dayOfTheWeek(self, day: int, month: int, year: int) -> str:
        cnt = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30]
        ans = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday","Sunday"]
        days = 0
        for i in range(1971,year):
            days += 366 if (i % 400 == 0 or (i % 4 == 0 and i % 100 != 0)) else 365
        for i in range(0,month-1):
            days += cnt[i]
        if month > 2 and (year % 400 == 0 or (year % 4  == 0 and year % 100 != 0)):
            days += 1
        days += day
        return ans[(days + 3) % 7]

File: Leetcode/JS/Leetcode1185-一周中的第几天.js
var dayOfTheWeek = function(day, month, year) {
    const ans = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday","Sunday"];
    const cnt = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30]
    let days = 0;
    for(let i = 1971;i < year;i++){
        let isleap = (i % 400 == 0 || (i % 4 == 0 && i % 100 != 0));
        days += isleap ? 366 : 365;
    }
    for(let i = 0;i < month - 1;i++) days += cnt[i];
    if((year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)) && month > 2) days += 1;
    days += day;
    return ans[(days + 3) % 7];
};